website/src/components/expandable/index.js

```diff
@@ -59,20 +62,35 @@ function Expandable({ children, alt_header = null, lifecycle }) {
     }, 3000);
   };
 
+  // Auto-expand when linked via hash (URL fragment)
   useEffect(() => {
     if (window.location.hash === `#${anchorId}`) {
-      setOn(true);
-      const element = document.getElementById(anchorId);
-      if (element) {
-        element.scrollIntoView({ behavior: 'smooth' });
-      }
+      setIsOpen(true);
+      detailsRef.current?.scrollIntoView({ behavior: 'smooth' });
     }
   }, [anchorId]);
 
+  // Observe search highlight and auto-expand
+  useEffect(() => {
+    const observer = new MutationObserver(() => {
+      const details = detailsRef.current;
+      if (details && details.querySelector('mark')) {
+        details.open = true;
+      }
+    });
+
+    observer.observe(document.body, {
+      childList: true,
+      subtree: true,
+    });
+
+    return () => observer.disconnect();
+  }, []);
+
   return (
-    <div id={anchorId} className={`${styles.expandableContainer}`}>
-      <div className={styles.header} onClick={handleToggleClick}>
-        <span className={`${styles.toggle} ${isOn ? styles.toggleDown : styles.toggleRight}`}></span>
+    <details ref={detailsRef} id={anchorId} className={styles.expandableContainer} open={isOpen}>
```

Comment:
Nice work setting the `open` attribute programmatically here.

Once small adjustment we need to make is adjusting the CSS for the chevron when ctrl + f searching.

You'll see in the screenshot even though the expandable is open, the chevron still is in the closed state.
![About_the_dbt_Cloud_IDE___dbt_Developer_Hub](https://github.com/user-attachments/assets/b5f4b5d2-e24b-49af-bf7e-7e5a9d8b94b0)

We should be able to clean this up by reading the `open` attribute within the CSS. 

For example, 
`details[open] .toggle {transform: rotate(225deg);}`

Reply:
oh great catch! thank you so much @john-rock ! ok updated it now and used `::before` since the chevron arrow is rendered through a pseudo-element.